src/main/java/com/spotify/autoscaler/Autoscaler.java

```diff
@@ -98,30 +90,37 @@ private void runUnsafe() {
             .filter(db::updateLastChecked)
             .map(
                 cluster ->
-                    CompletableFuture.runAsync(() -> runForCluster(cluster), executorService))
+                    CompletableFuture.runAsync(
+                        () -> runForCluster(cluster, hasRun), executorService))
             .toArray(CompletableFuture[]::new);
 
     CompletableFuture.allOf(futures).join();
   }
 
-  private void runForCluster(final BigtableCluster cluster) {
+  private void runForCluster(
+      final BigtableCluster cluster, final ConcurrentHashMap<String, Boolean> hasRun) {
     BigtableUtil.pushContext(cluster);
-    logger.info("Autoscaling cluster!");
-    try (final BigtableSession session = sessionProvider.apply(cluster);
-        final AutoscaleJob job =
-            autoscaleJobFactory.createAutoscaleJob(
-                session, () -> stackDriverClient, cluster, db, autoscalerMetrics, Instant::now)) {
-      job.run();
+    LOGGER.info("Autoscaling cluster!");
+    BigtableSession session = null;
+    try {
```

Comment:
I guess if we delete the `hasRun` bit, this can still be a try-with-resources block.